### PR TITLE
fix subversion test.

### PR DESCRIPTION
**Pull request by jtnord**
The subversion container could no longer be built as the ubuntu version
was no longer supported and its repos removed.

Switched to Debain (from ubuntu) as this contained packages for
everything that was required.

Using ViewVC not webSVN now - but nothing was actually using this (we
could remove it entierly if we want.

took the opertunity to cleanup the docker file

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jtnord** [comment]
Works locally, but tests that use SVN revisions fail in CI.  Needs more work ☹️

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SvnContainer/Dockerfile [hunk 1]
```diff
@@ -1,75 +1,59 @@
...
-FROM ubuntu:precise
-RUN apt-get update && apt-get install -y wget
-
-# Install Subversion 1.8 and Apache
-RUN echo 'deb http://archive.ubuntu.com/ubuntu precise main universe' >> /etc/apt/sources.list
-RUN sh -c 'echo "deb http://opensource.wandisco.com/ubuntu precise svn18" >> /etc/apt/sources.list.d/WANdisco.list'
-RUN wget -q http://opensource.wandisco.com/wandisco-debian.gpg -O- | apt-key add -
-RUN apt-get update -y
-RUN apt-get install -y subversion apache2 libapache2-svn
+FROM debian:buster
```
Comment (jtnord) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SvnContainer/Dockerfile:2]: ubuntu has no package for `viewvc` or `viewsvn`.  We do not actually use this currently and I thought about removing it, but for now I decided to leave it in.

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SvnContainer/config/supervisord.conf [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 nodaemon=true
+user=root
 
 [program:apache2]
-command=/bin/bash -c "exec /usr/sbin/apache2 -DFOREGROUND"
+command=/bin/bash -c "source /etc/apache2/envvars && /usr/sbin/apache2 -DFOREGROUND"
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
```
Comment (jtnord) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SvnContainer/config/supervisord.conf:10]: this tells supervisor to not put the output from the commands into a separate file but use sys.out sys.err (which will then be visible in `docker log`)

File: src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SvnContainer/config/viewvc.conf [hunk 3]
```diff
@@ -0,0 +1,1206 @@
...
+## disk, or grant new names to the clashing item in cvs_roots or
+## svn_roots.  Each parent path is processed sequentially, so the
+## names of repositories under later parent paths may override earlier
+## ones.
+##
+## Example:
+## root_parents = /opt/svn: svn,
+##                /opt/cvs: cvs
+##
+root_parents = /var/lib/svn: svn
```
Comment (jtnord) [inline src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/SvnContainer/config/viewvc.conf:157]: this was all I changed in the default config file

**Review by bmunozm — APPROVED** [review]

**Comment by jtnord** [comment]
passing in the `SubversionPluginTests` but failing in `plugins.WorkflowPluginTest#subversion` with

```
2021-06-04 13:05:16,519 INFO Set uid to user 0 succeeded
2021-06-04 13:05:16,520 INFO supervisord started with pid 1
2021-06-04 13:05:17,522 INFO spawned: 'svnserve' with pid 9
2021-06-04 13:05:17,524 INFO spawned: 'ssh' with pid 10
2021-06-04 13:05:17,525 INFO spawned: 'apache2' with pid 11
Server listening on 0.0.0.0 port 22.
Server listening on :: port 22.
[Fri Jun 04 13:05:17.552862 2021] [so:warn] [pid 11] AH01574: module dav_module is already loaded, skipping
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.3. Set the 'ServerName' directive globally to suppress this message
httpd (pid 9) already running
2021-06-04 13:05:17,567 INFO exited: apache2 (exit status 0; not expected)
2021-06-04 13:05:18,568 INFO success: svnserve entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2021-06-04 13:05:18,568 INFO success: ssh entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2021-06-04 13:05:18,569 INFO spawned: 'apache2' with pid 12
[Fri Jun 04 13:05:18.598125 2021] [so:warn] [pid 12] AH01574: module dav_module is already loaded, skipping
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.3. Set the 'ServerName' directive globally to suppress this message
httpd (pid 9) already running
2021-06-04 13:05:18,613 INFO exited: apache2 (exit status 0; not expected)
2021-06-04 13:05:20,616 INFO spawned: 'apache2' with pid 13
[Fri Jun 04 13:05:20.641513 2021] [so:warn] [pid 13] AH01574: module dav_module is already loaded, skipping
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.3. Set the 'ServerName' directive globally to suppress this message
httpd (pid 9) already running
2021-06-04 13:05:20,655 INFO exited: apache2 (exit status 0; not expected)
2021-06-04 13:05:23,660 INFO spawned: 'apache2' with pid 14
[Fri Jun 04 13:05:23.686608 2021] [so:warn] [pid 14] AH01574: module dav_module is already loaded, skipping
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.3. Set the 'ServerName' directive globally to suppress this message
httpd (pid 9) already running
2021-06-04 13:05:23,700 INFO exited: apache2 (exit status 0; not expected)
2021-06-04 13:05:24,701 INFO gave up: apache2 entered FATAL state, too many start retries too quickly
```

**Comment by jtnord** [comment]
> I am not exactly sure why apache2 fails to launch as svnserve (pid9) is running :-/

I do now - fix incoming...

**Pull request merged**